### PR TITLE
Basic model to calculate expiry times

### PR DESCRIPTION
**Pull request by luckysori**
Resolves comit-network/RFCs#14.

The actual model was moved to overleaf because it's easier to write math: https://www.overleaf.com/read/sswbyytnfjhy

DM @luckysori or @LLFourn for editable link.

Note that we don't now have actual code to calculate times from the model. This will be done this quarter in a follow up task.

### CODE REVIEW

File: 0001-basic-expiry-model.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+## Context
+
+The protocol defined in [RFC003](https://github.com/comit-network/RFCs/blob/master/RFC-003-SWAP-basic.md) introduces the concept of expiry times, which mark the time when assets can be refunded to their original owners. Values that are reasonable and that ensure that the protocol is secure for both parties must be estimated.
+
+## Research
+
+### Findings
+
+@LLFourn and I had a couple of meetings and we figured out some things:
```
Comment (D4nte) [inline 0001-basic-expiry-model.md]:
```suggestion
Here are the results of initial discussions:
```

File: 0001-basic-expiry-model.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+- The difference between `alpha_expiry` and `beta_expiry` has an effect on the security of the model for *Bob*. This does not affect Alice since she will have already redeemed `beta_asset` by the time this is relevant.
+- `beta_expiry` has an effect on the time that *Alice* will have to redeem `beta_asset` and therefore on the likelihood that the protocol will be completed successfully. This does _not_ have any direct effect on the security of the protocol, but rather on its *execution guarantees*.
```
Comment (D4nte) [inline 0001-basic-expiry-model.md]:
```suggestion
- `beta_expiry` has an effect on the time that *Alice* will have to redeem `beta_asset` and therefore on the likelihood that the protocol will be completed successfully. This does _not_ have any direct effect on the security of the protocol, but rather on its *execution guarantee*.
```

File: 0001-basic-expiry-model.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+- Block time on `alpha_ledger` can be modelled as a random variable that follows an exponential distribution with parameter <img src="https://latex.codecogs.com/gif.latex?\lambda"/>.
+- A confirmation is given by the arrival of a new block.
+- The value of `beta_expiry`, <img src="https://latex.codecogs.com/gif.latex?E_{\beta}"/>, is constant.
+
+#### Model
+
+To determine `alpha_expiry`, the time <img src="https://latex.codecogs.com/gif.latex?\Delta_{R}"/> Bob will need to get his desired number of confirmations on his redeem transaction of `alpha_asset` will need to be estimated.
+
+The time it would take for Bob's redeem transaction to obtain <img src="https://latex.codecogs.com/gif.latex?k"/> confirmations on `alpha_ledger` can be modelled as a sum of <img src="https://latex.codecogs.com/gif.latex?k"/> mutually independent exponential random variables with the same rate parameter <img src="https://latex.codecogs.com/gif.latex?\lamda"/>, which is equal to an <img src="https://latex.codecogs.com/gif.latex?\textrm{Erlang}(k,\lambda)"/> distribution.
```
Comment (D4nte) [inline 0001-basic-expiry-model.md]:
```suggestion
The time it would take for Bob's redeem transaction to obtain <img src="https://latex.codecogs.com/gif.latex?k"/> confirmations on `alpha_ledger` can be modelled as a sum of <img src="https://latex.codecogs.com/gif.latex?k"/> mutually independent exponential random variables with the same rate parameter <img src="https://latex.codecogs.com/gif.latex?\lambda"/>, which is equal to an <img src="https://latex.codecogs.com/gif.latex?\textrm{Erlang}(k,\lambda)"/> distribution.
```

File: 0001-basic-expiry-model.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+Letting <img src="https://latex.codecogs.com/gif.latex?C_{k}\sim\textrm{Erlang}(k,\lambda)"/>,
+
+<img src="https://latex.codecogs.com/gif.latex?E_{\beta}=t_{0}&plus;T^{\alpha}_{A}&plus;T^{\beta}_{B}&plus;T^{\beta}_{A}&plus;Q_{C_{k}}(p)"/>, <img src="https://latex.codecogs.com/gif.latex?(1)"/>
+
+where <img src="https://latex.codecogs.com/gif.latex?Q_{C_{k}}(p)"/> is the [quantile function](https://en.wikipedia.org/wiki/Quantile_function) of <img src="https://latex.codecogs.com/gif.latex?C_{k}"/> for a probability <img src="https://latex.codecogs.com/gif.latex?p"/>. This is, a function returning the minimum time taken for <img src="https://latex.codecogs.com/gif.latex?C_{k}"/> to take place with probability <img src="https://latex.codecogs.com/gif.latex?p"/>.
+
+As mentioned in the previous section, quantile functions for Erlang random variables can only be approximated via [numerical methods](https://docs.scipy.org/doc/scipy-0.16.1/reference/generated/scipy.stats.erlang.html).
+
+
+With this knowledge, `beta_expiry` can be estimated using <img src="https://latex.codecogs.com/gif.latex?(1)"/>.
```
Comment (D4nte) [inline 0001-basic-expiry-model.md]:
```suggestion
With this knowledge, `beta_expiry` can be estimated using [1].
```

File: 0001-basic-expiry-model.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+#### Model
+
+To determine `beta_expiry`, the random variable <img src="https://latex.codecogs.com/gif.latex?R_{\beta}"/>, which models the time Alice will need to get her desired number of confirmations on her redeem transaction of `beta_asset`, will need to be approximated.
+
+The time it would take for Alice's redeem transaction to obtain <img src="https://latex.codecogs.com/gif.latex?k"/> confirmations on `beta_ledger` can be modelled as a sum of <img src="https://latex.codecogs.com/gif.latex?k"/> mutually independent exponential random variables with the same rate parameter <img src="https://latex.codecogs.com/gif.latex?\lamda"/>, which is equal to an <img src="https://latex.codecogs.com/gif.latex?\textrm{Erlang}(k,\lambda)"/> distribution.
+
+Letting <img src="https://latex.codecogs.com/gif.latex?C_{k}\sim\textrm{Erlang}(k,\lambda)"/>,
+
+<img src="https://latex.codecogs.com/gif.latex?E_{\beta}=t_{0}&plus;T^{\alpha}_{A}&plus;T^{\beta}_{B}&plus;T^{\beta}_{A}&plus;Q_{C_{k}}(p)"/>, <img src="https://latex.codecogs.com/gif.latex?(1)"/>
```
Comment (D4nte) [inline 0001-basic-expiry-model.md]:
```suggestion
<img src="https://latex.codecogs.com/gif.latex?E_{\beta}=t_{0}&plus;T^{\alpha}_{A}&plus;T^{\beta}_{B}&plus;T^{\beta}_{A}&plus;Q_{C_{k}}(p)"/>, [1]
```

**Review by D4nte — COMMENTED** [review]
Will have to re-read the maths later.

File: 0001-basic-expiry-model.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+where <img src="https://latex.codecogs.com/gif.latex?Q_{C_{k}}(p)"/> is the [quantile function](https://en.wikipedia.org/wiki/Quantile_function) of <img src="https://latex.codecogs.com/gif.latex?C_{k}"/> for a probability <img src="https://latex.codecogs.com/gif.latex?p"/>. This is, a function returning the minimum time taken for <img src="https://latex.codecogs.com/gif.latex?C_{k}"/> to take place with probability <img src="https://latex.codecogs.com/gif.latex?p"/>.
```
Comment (luckysori) [inline 0001-basic-expiry-model.md]: :x: Missing introduction of t_{0}.

File: 0001-basic-expiry-model.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+
+#### Assumptions
+
+- Alice provides <img src="https://latex.codecogs.com/gif.latex?T^{\alpha}_{A}"/>, the time in which she is overwhelmingly confident she will be able to confirm a transaction on `alpha_ledger`. She has direct influence over this since she can choose the fee.
+- Bob provides <img src="https://latex.codecogs.com/gif.latex?T^{\beta}_{B}"/>, the time in which he is overwhelmingly confident he will be able to confirm a transaction on `beta_ledger`. He has direct influence over this since he can choose the fee.
+- Alice provides <img src="https://latex.codecogs.com/gif.latex?T^{\beta}_{A}"/>, the time in which she is overwhelmingly confident she will be able to confirm a transaction on `beta_ledger`. She has direct influence over this since she can choose the fee.
+- Alice provides <img src="https://latex.codecogs.com/gif.latex?k"/>, the number of additional confirmations she would need to have on her redeem transaction before she considers `beta_asset` to be her own.
+- Block time on `alpha_ledger` can be modelled as a random variable that follows an exponential distribution with parameter <img src="https://latex.codecogs.com/gif.latex?\lambda"/>.
```
Comment (luckysori) [inline 0001-basic-expiry-model.md]: :x:
```suggestion
- Block time on `beta_ledger` can be modelled as a random variable that follows an exponential distribution with parameter <img src="https://latex.codecogs.com/gif.latex?\lambda"/>.
```

**Review by thomaseizinger — COMMENTED** [review]
I am seriously impressed by your work guys!

Very nicely done :)

File: 0001-basic-expiry-model.md [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+where <img src="https://latex.codecogs.com/gif.latex?t_{0}"/> is the time when all the protocol parameters are set and <img src="https://latex.codecogs.com/gif.latex?Q_{C_{k}}(p)"/> is the [quantile function](https://en.wikipedia.org/wiki/Quantile_function) of <img src="https://latex.codecogs.com/gif.latex?C_{k}"/> for a probability <img src="https://latex.codecogs.com/gif.latex?p"/>.
+This is, a function returning the minimum time taken for <img src="https://latex.codecogs.com/gif.latex?C_{k}"/> to take place with probability <img src="https://latex.codecogs.com/gif.latex?p"/>.
+
+As mentioned in the previous section, quantile functions for Erlang random variables can only be approximated via [numerical methods](https://docs.scipy.org/doc/scipy-0.16.1/reference/generated/scipy.stats.erlang.html).
+
+
+With this knowledge, `beta_expiry` can be estimated using [1].
+
+### Example
+
```
Comment (luckysori) [inline 0001-basic-expiry-model.md:25]:
```suggestion

All times expressed in minutes.

```

File: 0001-basic-expiry-model.md [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+where <img src="https://latex.codecogs.com/gif.latex?Q_{C_{k}}(p)"/> is the [quantile function](https://en.wikipedia.org/wiki/Quantile_function) of <img src="https://latex.codecogs.com/gif.latex?C_{k}"/> for a probability <img src="https://latex.codecogs.com/gif.latex?p"/>. 
+This is, a function returning the minimum time taken for <img src="https://latex.codecogs.com/gif.latex?C_{k}"/> to take place with probability <img src="https://latex.codecogs.com/gif.latex?p"/>.
+
+Quantile functions for Erlang random variables can only be approximated via [numerical methods](https://docs.scipy.org/doc/scipy-0.16.1/reference/generated/scipy.stats.erlang.html).
+
+Knowing all this, `alpha_expiry` can be estimated:
+
+<img src="https://latex.codecogs.com/gif.latex?E_{\alpha}=E_{\beta}&plus;T_{\alpha}&plus;Q_{C_{k}}(p)"/>,
+
+where the value of <img src="https://latex.codecogs.com/gif.latex?p"/> is directly proportional to the security of the model (and the size of `alpha_expiry`).
```
Comment (LLFourn) [inline 0001-basic-expiry-model.md]: What does this statement mean? 𝑃 is proportional to "the security of the model"?

File: 0001-basic-expiry-model.md [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+He has direct influence over this since he can choose the fee.
+- Alice provides <img src="https://latex.codecogs.com/gif.latex?T^{\beta}_{A}"/>, the time in which she is overwhelmingly confident she will be able to confirm a transaction on `beta_ledger`.
+She has direct influence over this since she can choose the fee.
+- Alice provides <img src="https://latex.codecogs.com/gif.latex?k"/>, the number of additional confirmations she would need to have on her redeem transaction before she considers `beta_asset` to be her own.
+- Block time on `beta_ledger` can be modelled as a random variable that follows an exponential distribution with parameter <img src="https://latex.codecogs.com/gif.latex?\lambda"/>.
+- A confirmation is given by the arrival of a new block.
+
+#### Model
+
+To determine `beta_expiry`, the random variable <img src="https://latex.codecogs.com/gif.latex?R_{\beta}"/>, which models the time Alice will need to get her desired number of confirmations on her redeem transaction of `beta_asset`, will need to be approximated.
```
Comment (LLFourn) [inline 0001-basic-expiry-model.md]: 🔧 This was too much passive voice for me
```suggestion
To determine `beta_expiry`, we need to approximate the random variable <img src="https://latex.codecogs.com/gif.latex?R_{\beta}"/>, the time Alice will need to get her desired number of confirmations on her redeem transaction of `beta_asset`
```

File: 0001-basic-expiry-model.md [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+
+The time it would take for Alice's redeem transaction to obtain <img src="https://latex.codecogs.com/gif.latex?k"/> confirmations on `beta_ledger` can be modelled as a sum of <img src="https://latex.codecogs.com/gif.latex?k"/> mutually independent exponential random variables with the same rate parameter <img src="https://latex.codecogs.com/gif.latex?\lamda"/>, which is equal to an <img src="https://latex.codecogs.com/gif.latex?\textrm{Erlang}(k,\lambda)"/> distribution.
+
+Letting <img src="https://latex.codecogs.com/gif.latex?C_{k}\sim\textrm{Erlang}(k,\lambda)"/>,
```
Comment (LLFourn) [inline 0001-basic-expiry-model.md]: Using λᵦ and λₐ would make this clearer.

File: 0001-basic-expiry-model.md [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+# Time in which Bob is overwhelmingly confident he will be able to confirm an Ethereum (beta ledger) transaction
+bob_beta_time = 2
+# Time in which Alice is overwhelmingly confident she will be able to confirm an Ethereum (beta ledger) transaction
+alice_beta_time = 1.5
+# Number of additional confirmations Alice needs on her redeem transaction of beta_asset
+alice_confirmations = 40
+# Ethereum (beta ledger) block time
+beta_block_time = 0.25
+# Time taken for k blocks to be generated on Ethereum (beta ledger) with probability 0.95
+time_for_confirmations_beta = erlang.ppf(0.999999, alice_confirmations, 0, beta_block_time)
```
Comment (LLFourn) [inline 0001-basic-expiry-model.md]: 🔧 
1. This comment is stale :)
2. extract the probability into a variable
3. We want Bob to lose money with much smaller probability than the trade being aborted. (e.g. 1/1_000 vs 1_000_000).

File: 0001-basic-expiry-model.md (hunk 9 above)
Comment (luckysori) [reply 0001-basic-expiry-model.md]: Yeah, I don't like that statement either.

File: 0001-basic-expiry-model.md (hunk 10 above)
Comment (luckysori) [reply 0001-basic-expiry-model.md]: We may disagree on this, but I generally prefer the passive voice for maths. Since I wrote it in the passive voice already, I'd like to keep it consistent.

File: 0001-basic-expiry-model.md (hunk 12 above)
Comment (luckysori) [reply 0001-basic-expiry-model.md]: Very good points!

File: 0001-basic-expiry-model.md [hunk 13]
```diff
@@ -0,0 +1,166 @@
...
+The protocol defined in [RFC003](https://github.com/comit-network/RFCs/blob/master/RFC-003-SWAP-basic.md) introduces the concept of expiry times, which mark the time when assets can be refunded to their original owners.
+Values that are reasonable and that ensure that the protocol is secure for both parties must be estimated.
+
+## Research
+
+### Findings
+
+Here are the results of initial discussions:
+
+- The difference between `alpha_expiry` and `beta_expiry` has an effect on the security of the model for *Bob*.
```
Comment (bonomat) [inline 0001-basic-expiry-model.md]: ❌ I don't agree with this sentence: Alice redeems `beta_asset` and hence `beta_expiry` has an effect on her security model.

File: 0001-basic-expiry-model.md [hunk 14]
```diff
@@ -0,0 +1,166 @@
...
+This does not affect Alice since she will have already redeemed `beta_asset` by the time this is relevant.
```
Comment (bonomat) [inline 0001-basic-expiry-model.md]: ❓By _when_ is _what_ relevant?

File: 0001-basic-expiry-model.md [hunk 15]
```diff
@@ -0,0 +1,166 @@
...
+- `beta_expiry` has an effect on the time that *Alice* will have to redeem `beta_asset` and therefore on the likelihood that the protocol will be completed successfully.
+This does _not_ have any direct effect on the security of the protocol, but rather on its *execution guarantees*.
```
Comment (bonomat) [inline 0001-basic-expiry-model.md]: ❌ We might want to define what _security of the protocol_ means in this case because if `bet_expiry` is too small, Alice could lose all of her funds.

File: 0001-basic-expiry-model.md [hunk 16]
```diff
@@ -0,0 +1,166 @@
...
+
+1. How long Bob will (at least) have to redeem `alpha_asset` if Alice redeems `beta_asset`.
+2. How long Alice will have to wait to be able to refund `alpha_asset` if the protocol isn't carried out to completion.
+
+It would therefore be ideal to find a value of `alpha_expiry` that lets Bob redeem `alpha_asset` with high probability while also ensuring Alice doesn't have to wait too long to be able to be refunded if it comes to that.
+
+#### Assumptions
+
+- Bob provides <img src="https://latex.codecogs.com/gif.latex?T_{\alpha}"/>, the time in which he is overwhelmingly confident he will be able to confirm a transaction on `alpha_ledger`.
+For simplicity, the model assumes that the probability of this happening within the time provided is 1.
```
Comment (bonomat) [inline 0001-basic-expiry-model.md]: ❌ In my opinion this is the most important part of the model which should be considered. Bob will never have a probability of 1 (100%), even at time . Additionally, if Bob waits too long then this probability to redeem the `alpha_asset` slowly decreases towards 0.

File: 0001-basic-expiry-model.md [hunk 17]
```diff
@@ -0,0 +1,166 @@
...
+- Bob provides <img src="https://latex.codecogs.com/gif.latex?k"/>, the number of additional confirmations he would need to have on his redeem transaction before he considers `alpha_asset` to be his own.
```
Comment (bonomat) [inline 0001-basic-expiry-model.md]: ❌ I don't think this really matters. The number of confirmations of funding `alpha_asset` are important, i.e. Bob wants to be sure that Alice can not double spend `alpha_asset`.

File: 0001-basic-expiry-model.md [hunk 18]
```diff
@@ -0,0 +1,166 @@
...
+- Block time on `alpha_ledger` can be modelled as a random variable that follows an exponential distribution with parameter <img src="https://latex.codecogs.com/gif.latex?\lambda_{\alpha}"/>.
+- A confirmation is given by the arrival of a new block.
+- The value of `beta_expiry`, <img src="https://latex.codecogs.com/gif.latex?E_{\beta}"/>, is constant.
```
Comment (bonomat) [inline 0001-basic-expiry-model.md]: 🔧 For making this model more complete we would need to add some more variables: 
* The amount of the HTLC matters, i.e. no one will attempt to double spend a transaction worth 5$
* The congestion of the network should be considered which has a direct influence to
* the transaction cost/fee. This should be considered and is itself an exponential growing function which gets higher the closer you get to the end of `*_expiry`.

File: 0001-basic-expiry-model.md [hunk 19]
```diff
@@ -0,0 +1,166 @@
...
+
+#### Model
+
+To determine `alpha_expiry`, the time <img src="https://latex.codecogs.com/gif.latex?\Delta_{R}"/> Bob will need to get his desired number of confirmations on his redeem transaction of `alpha_asset` will need to be estimated.
```
Comment (bonomat) [inline 0001-basic-expiry-model.md]: Can you rephrase this sentence please or introduce  first?

File: 0001-basic-expiry-model.md [hunk 20]
```diff
@@ -0,0 +1,166 @@
...
+- The difference between `alpha_expiry` and `beta_expiry` has an effect on the security of the model for *Bob*.
+This does not affect Alice since she will have already redeemed `beta_asset` by the time this is relevant.
+- `beta_expiry` has an effect on the time that *Alice* will have to redeem `beta_asset` and therefore on the likelihood that the protocol will be completed successfully.
+This does _not_ have any direct effect on the security of the protocol, but rather on its *execution guarantees*.
+
+Given this, the topic of this spike changed from creating a security model to, more generally, creating a basic model to calculate expiry times.
+
+### Alpha expiry
+
+There are two ways in which `alpha_expiry`, <img src="https://latex.codecogs.com/gif.latex?E_{\alpha}"/>, has an effect on the protocol:
```
Comment (bonomat) [inline 0001-basic-expiry-model.md]: I would rather see variable introduction in this form: 
```suggestion
There are two ways in which `alpha_expiry` (<img src="https://latex.codecogs.com/gif.latex?E_{\alpha}"/>) has an effect on the protocol:
```

File: 0001-basic-expiry-model.md [hunk 21]
```diff
@@ -0,0 +1,166 @@
...
+
+1. How long Bob will (at least) have to redeem `alpha_asset` if Alice redeems `beta_asset`.
+2. How long Alice will have to wait to be able to refund `alpha_asset` if the protocol isn't carried out to completion.
+
+It would therefore be ideal to find a value of `alpha_expiry` that lets Bob redeem `alpha_asset` with high probability while also ensuring Alice doesn't have to wait too long to be able to be refunded if it comes to that.
+
+#### Assumptions
+
+- Bob provides <img src="https://latex.codecogs.com/gif.latex?T_{\alpha}"/>, the time in which he is overwhelmingly confident he will be able to confirm a transaction on `alpha_ledger`.
```
Comment (bonomat) [inline 0001-basic-expiry-model.md]: > Bob provides , the time in which he is overwhelmingly confident he will be able to confirm a transaction on `alpha_ledger`.

I'm not sure if _confidence_ should be part of a mathematical model. Bob's (or Alice's) confidence depends on the amount locked up and the chance of getting into a block in time which again depends on the congestion of the network and the transaction fee. We can use all those variables in the model.

File: 0001-basic-expiry-model.md [hunk 22]
```diff
@@ -0,0 +1,166 @@
...
+For simplicity, the model assumes that the probability of this happening within the time provided is 1.
+- Bob provides <img src="https://latex.codecogs.com/gif.latex?k"/>, the number of additional confirmations he would need to have on his redeem transaction before he considers `alpha_asset` to be his own.
+- Block time on `alpha_ledger` can be modelled as a random variable that follows an exponential distribution with parameter <img src="https://latex.codecogs.com/gif.latex?\lambda_{\alpha}"/>.
```
Comment (bonomat) [inline 0001-basic-expiry-model.md]: It would be worth mentioning an example here.

File: 0001-basic-expiry-model.md (hunk 13 above)
Comment (bonomat) [inline 0001-basic-expiry-model.md]: Should we introduce the type of `*_expiry`?

File: 0001-basic-expiry-model.md [hunk 23]
```diff
@@ -0,0 +1,166 @@
...
+- A confirmation is given by the arrival of a new block.
```
Comment (bonomat) [inline 0001-basic-expiry-model.md]: Is the chance of finding a block in particular time important? What if there are no blocks?

**Review by bonomat — COMMENTED** [review]
Good job on this model. I have a few comments but let's discuss it offline.

File: 0001-basic-expiry-model.md [hunk 24]
```diff
@@ -0,0 +1,166 @@
...
+
+where <img src="https://latex.codecogs.com/gif.latex?t_{0}"/> is the time when all the protocol parameters are set and <img src="https://latex.codecogs.com/gif.latex?Q_{C_{k}}(p_{\beta})"/> is the [quantile function](https://en.wikipedia.org/wiki/Quantile_function) of <img src="https://latex.codecogs.com/gif.latex?C_{k}"/> for a probability <img src="https://latex.codecogs.com/gif.latex?p_{\beta}"/>.
+This is, a function returning the minimum time taken for <img src="https://latex.codecogs.com/gif.latex?C_{k}"/> to take place with probability <img src="https://latex.codecogs.com/gif.latex?p_{\beta}"/>.
+
+As mentioned in the previous section, quantile functions for Erlang random variables can only be approximated via [numerical methods](https://docs.scipy.org/doc/scipy-0.16.1/reference/generated/scipy.stats.erlang.html).
+
+
+With this knowledge, `beta_expiry` can be estimated using [1]. In this case, <img src="https://latex.codecogs.com/gif.latex?p_{\beta}"/> represents the probability that Alice redeems `beta_asset` before `beta_expiry`.
+In other words, the probability that the protocol will not stop half-way through.
```
Comment (bonomat) [inline 0001-basic-expiry-model.md]: Even if Alice does not redeem, the protocol does not _stop_, it's just a different path (the refund path) which is being executed.

File: 0001-basic-expiry-model.md [hunk 25]
```diff
@@ -0,0 +1,166 @@
...
+For simplicity, the model assumes that the probability of this happening within the time provided is 1.
+- Alice provides <img src="https://latex.codecogs.com/gif.latex?T^{\beta}_{A}"/>, the time in which she is overwhelmingly confident she will be able to confirm a transaction on `beta_ledger`.
+For simplicity, the model assumes that the probability of this happening within the time provided is 1.
+- Alice provides <img src="https://latex.codecogs.com/gif.latex?k"/>, the number of additional confirmations she would need to have on her redeem transaction before she considers `beta_asset` to be her own.
+- Block time on `beta_ledger` can be modelled as a random variable that follows an exponential distribution with parameter <img src="https://latex.codecogs.com/gif.latex?\lambda_{\beta}"/>.
+- A confirmation is given by the arrival of a new block.
+
+#### Model
+
+To determine `beta_expiry`, the random variable <img src="https://latex.codecogs.com/gif.latex?R_{\beta}"/>, which models the time Alice will need to get her desired number of confirmations on her redeem transaction of `beta_asset`, will need to be approximated.
```
Comment (bonomat) [inline 0001-basic-expiry-model.md]:
```suggestion
To determine `beta_expiry` (<img src="https://latex.codecogs.com/gif.latex?E_{\beta}"/>), the random variable <img src="https://latex.codecogs.com/gif.latex?R_{\beta}"/>, which models the time Alice will need to get her desired number of confirmations on her redeem transaction of `beta_asset`, will need to be approximated.
```

File: 0001-basic-expiry-model.md (hunk 23 above)
Comment (bonomat) [reply 0001-basic-expiry-model.md]: Please link to erlang and exponential distribution.

File: 0001-basic-expiry-model.md (hunk 2 above)
Comment (D4nte) [reply 0001-basic-expiry-model.md]: the guarantee of its execution. Not sure why this is plural.

Comment (luckysori) [reply 0001-basic-expiry-model.md]: We use both ways in the original [OKR issue](https://github.com/comit-network/comit-rs/issues/670), but I prefer "execution guarantees". I asked @LLFourn to write the next draft anyway, so he can choose or just remove it.

File: 0001-basic-expiry-model.md [hunk 26]
```diff
@@ -0,0 +1,166 @@
+# Basic model to calculate expiry times
+
+* Status: proposed
```
Comment (luckysori) [inline 0001-basic-expiry-model.md]: :x: Please remove the status. See #3.

**Comment by D4nte** [comment]
Is this PR still valid? didn't @LLFourn moved the spike result somewhere else? Please clarify.

**Comment by LLFourn** [comment]
@D4nte The spike references the overleaf so yes. Please approve.

**Review by thomaseizinger — APPROVED** [review]
